accounts/models.py

```diff
@@ -0,0 +1,92 @@
+from allauth.account.signals import user_signed_up
+from django.dispatch import receiver
+from django.contrib.auth.models import User
+from django.db import models
+from django.conf import settings
+from django.db.models.signals import post_delete
+from .lists import user_types, lms_modules
+
+
+class Profile(models.Model):
+    """
+    Define Profile Model with OneToOne relationship to AUTH_USER_MODEL and
+    custom fields to suit Signup Form Extending in forms.py replicating same
+    in DB.
+    Using "related_name" in the OneToOne relationship between the two models
+    specifies the reverse relationship to the User Model, allowing us to
+    target the custom fields for injection into the profile.html
+    template via `{{ user.profile.<<field_name>> }}`.
+    """
+    user = models.OneToOneField(
+        settings.AUTH_USER_MODEL, default=1,
+        related_name='profile',
+        on_delete=models.CASCADE
+    )
+    slack_display_name = models.CharField(
+        max_length=80,
+        blank=False,
+        null=True
+    )
+    user_type = models.CharField(
+        max_length=20,
+        blank=False,
+        null=True,
+        choices=user_types
+    )
+    current_lms_module = models.CharField(
+        max_length=35,
+        blank=False,
+        null=True,
+        choices=lms_modules
+    )
+
+    def save(self, *args, **kwargs):
+        # when signup takes place
+        try:
+            self.slack_display_name = self.user.slack_display_name
+            self.user_type = self.user.user_type
+            self.current_lms_module = self.user.current_lms_module
+        # when saving via admin panel
+        except:
+            self.slack_display_name = self.user.profile.slack_display_name
+            self.user_type = self.user.profile.user_type
+            self.current_lms_module = self.user.profile.current_lms_module
+
+        super(Profile, self).save(*args, **kwargs)
+
+    def __str__(self):
+        return self.user.username
```

Comment:
If you are going with my suggestion to make username not required, I would change this to email.